src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -49,6 +52,75 @@ public static Encoding OutputEncoding
             get { return GetConsoleEncoding(); }
         }
 
+        private static readonly object InternalSyncObject = new object();
```

Comment:
> and why that's necessary.

Sorry I did not understand this? Do you mean why name it this way? How about s_stdInReaderSyncObject, to correctly depict the usage.